auth/src/test/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivityTest.java

```diff
@@ -0,0 +1,213 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.firebase.ui.auth.ui.idp;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.when;
+
+import android.content.Intent;
+import android.view.View;
+import android.widget.Button;
+import android.widget.LinearLayout;
+
+import com.firebase.ui.auth.AuthUI;
+import com.firebase.ui.auth.BuildConfig;
+import com.firebase.ui.auth.R;
+import com.firebase.ui.auth.provider.GoogleProvider;
+import com.firebase.ui.auth.test_helpers.ActivityHelperShadow;
+import com.firebase.ui.auth.test_helpers.AutoCompleteTask;
+import com.firebase.ui.auth.test_helpers.CustomRobolectricGradleTestRunner;
+import com.firebase.ui.auth.test_helpers.FacebookProviderShadow;
+import com.firebase.ui.auth.test_helpers.FakeAuthResult;
+import com.firebase.ui.auth.test_helpers.FirebaseAuthWrapperImplShadow;
+import com.firebase.ui.auth.test_helpers.GoogleProviderShadow;
+import com.firebase.ui.auth.test_helpers.TestConstants;
+import com.firebase.ui.auth.test_helpers.TestHelper;
+import com.firebase.ui.auth.ui.ExtraConstants;
+import com.firebase.ui.auth.ui.account_link.SaveCredentialsActivity;
+import com.firebase.ui.auth.ui.email.EmailHintContainerActivity;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FacebookAuthProvider;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.auth.GoogleAuthProvider;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.Shadows;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(CustomRobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class,
+        shadows = {
+                FirebaseAuthWrapperImplShadow.class,
+                GoogleProviderShadow.class,
+                FacebookProviderShadow.class
+        }, sdk = 21)
+public class AuthMethodPickerActivityTest {
+
+    @Before
+    public void setUp() {}
+
+    @Test
+    public void testAllProvidersArePopulated() {
+        List<String> providers = Arrays.asList(
+                AuthUI.FACEBOOK_PROVIDER,
+                AuthUI.GOOGLE_PROVIDER,
+                AuthUI.EMAIL_PROVIDER);
+        new GoogleProvider(null, null);
```

Comment:
It was there to initialize the mocks, but isn't actually needed in this case.  I will remove it.